src/main/kotlin/com/openlattice/hazelcast/serializers/CollectionTemplatesStreamSerializer.kt

```diff
@@ -0,0 +1,74 @@
+package com.openlattice.hazelcast.serializers
+
+import com.google.common.collect.Maps
+import com.hazelcast.nio.ObjectDataInput
+import com.hazelcast.nio.ObjectDataOutput
+import com.kryptnostic.rhizome.pods.hazelcast.SelfRegisteringStreamSerializer
+import com.openlattice.collections.CollectionTemplates
+import com.openlattice.hazelcast.StreamSerializerTypeIds
+import org.springframework.stereotype.Component
+import java.util.*
+import java.util.concurrent.ConcurrentMap
+
+@Component
+class CollectionTemplatesStreamSerializer : SelfRegisteringStreamSerializer<CollectionTemplates> {
+
+    companion object {
+
+        fun serialize(out: ObjectDataOutput, `object`: CollectionTemplates) {
+            out.writeInt(`object`.templates.size)
+
+            `object`.templates.forEach {
+                UUIDStreamSerializer.serialize(out, it.key)
+                out.writeInt(it.value.size)
+
+                it.value.forEach  {
```

Comment:
Consider the following instead:
```
out.writeInt(`object`.templates.size)
`object`.templates.forEach {
    UUIDStreamSerializer.serialize(out, it.key)
    it.value.forEach {
        UUIDStreamSerializer.serialize(out, it.key)
        val subMap = it.value
        SetStreamSerializers.fastUUIDSetSerialize(out, subMap.keys)
        SetStreamSerializers.fastUUIDSetSerialize(out, subMap.values)
    }
}
```
and
```
val size = `in`.readInt()
val templates = Maps.newConcurrentMap<UUID, ConcurrentMap<UUID ,UUID>>()
for (i in 0 until size) {
    val key = UUIDStreamSerializer.deserialize(`in`)

    val newMap = Maps.newConcurrentMap<UUID, UUID>()
    val keys = SetStreamSerializers.fastUUIDSetDeserialize(`in`).iterator()
    val vals = SetStreamSerializers.fastUUIDSetDeserialize(`in`).iterator()
    while (keys.hasNext()) {
         newMap[keys.next()] = vals.next()
    }
}
```

Reply:
Also maybe there should be a UUIDMapStreamSerializer that does this because it feels scary to rewrite this ever